### ISSUE TITLE
questions about the result

### ISSUE THREAD

**Issue by qianzhang2018** [tianzhi0549/FCOS#82]
Thanks for the great work.
 When I fit the code to my dataset, i have some puzzles.
1.There will be omissions in the results，but I think this situation should be easy to detect？
![webwxgetmsgimg](https://user-images.githubusercontent.com/41365581/60781240-245f1480-a174-11e9-84f0-bedeef70f787.jpeg)
2.I found that the boxes would overlap.This problem occurs many times in the result.
this code use nms，doesn‘t it？
![1](https://user-images.githubusercontent.com/41365581/60781352-9b94a880-a174-11e9-8f42-e3f4708a9538.png)
Any advice will be appreciated, thanks

**Comment by tianzhi0549** [comment]
@qianzhang2018
1) Can you try to use lower score_thresholds? The code is at https://github.com/tianzhi0549/FCOS/blob/eaca87d53048879770d2b50858dda9f0d98f6609/demo/fcos_demo.py#L57

2) NMS is only applied to the bounding-boxes with the same class.

**Comment by qianzhang2018** [comment]
Thank you for your reply. i set " thresholds_for_classes = [
        0.05, 0.05, 0.05,0.05, 0.05, 0.05,0.05, 0.05, 0.05,0.05, 0.05, 
    ]"（i have 10 classes）
but the result is same ,i think the network doesn't detect it?
i train on one 1080,batch_size =2 ,BASE_LR: 0.0001 # 0.01 。
Is it possible that it's a training problem?

**Comment by tianzhi0549** [comment]
@qianzhang2018 Yes, it's possible. Your batch size is too small. I suggest that you use batch size >= 16.

**Comment by qianzhang2018** [comment]
thank you ,I'll do more experiments in it .

**Comment by hello-piger** [comment]
> Thanks for the great work.
> When I fit the code to my dataset, i have some puzzles.
> 1.There will be omissions in the results，but I think this situation should be easy to detect？
> ![webwxgetmsgimg](https://user-images.githubusercontent.com/41365581/60781240-245f1480-a174-11e9-84f0-bedeef70f787.jpeg)
> 2.I found that the boxes would overlap.This problem occurs many times in the result.
> this code use nms，doesn‘t it？
> ![1](https://user-images.githubusercontent.com/41365581/60781352-9b94a880-a174-11e9-8f42-e3f4708a9538.png)
> Any advice will be appreciated, thanks
hello! did you use the kitti dataset?and did you transform the dataset into coco style?

**Comment by hello-piger** [comment]
hello! did you use the kitti dataset? have you transform the dataset into coco style?

**Comment by hello-piger** [comment]
> Thanks for the great work.
> When I fit the code to my dataset, i have some puzzles.
> 1.There will be omissions in the results，but I think this situation should be easy to detect？
> ![webwxgetmsgimg](https://user-images.githubusercontent.com/41365581/60781240-245f1480-a174-11e9-84f0-bedeef70f787.jpeg)
> 2.I found that the boxes would overlap.This problem occurs many times in the result.
> this code use nms，doesn‘t it？
> ![1](https://user-images.githubusercontent.com/41365581/60781352-9b94a880-a174-11e9-8f42-e3f4708a9538.png)
> Any advice will be appreciated, thanks

**Comment by hello-piger** [comment]
hello! did you use the kitti dataset? have you transform the dataset into coco style?

**Comment by qianzhang2018** [comment]
i don’t use the kitti.i only change my dataset into coco style



---Original---

**Comment by dreamhighchina** [comment]
@tianzhi0549 
 thresholds_for_classes = [ 
NMS is only applied to the bounding-boxes with the same class.

how can i change the NMS with all class？？i did not find NMS in your code。。。
